experimental/GroebnerWalk/src/markedGB.jl

```diff
@@ -0,0 +1,121 @@
+using Oscar 
+import Oscar: gens, base_ring
+import Base: length, getindex, setindex!
+
+#constructs a "markedGB" consisting of two lists 
+#gens, a vector of generators of a Gröbner basis
+#markings, a vector "Markings" of G, corresponding to leading terms w.r.t some monomial ordering
+#TODO: maybe initialize s.t. G is reduced/monic? 
+
+struct MarkedGroebnerBasis
+    gens::Vector{<:MPolyRingElem}
+    markings::Vector{<:MPolyRingElem}
+
+    function MarkedGroebnerBasis(gens::Vector{<:MPolyRingElem}, markings::Vector{<:MPolyRingElem})
+        if length(gens) != length(markings)
+            throw(ArgumentError("Inputs are of different length"))
+        else
```

Comment:
There is `@req` to make easier input sanitation (see https://docs.oscar-system.org/dev/Hecke/manual/developer/macros/#@req)